src/app/screens/LoginScreen/index.tsx

```diff
@@ -42,40 +43,43 @@ function LoginScreen() {
   const onPress = () => {
     dispatch(actionCreators.logIn(userValue, passwordValue));
   };
+  const loginLoading = useSelector((state: AppState) => state.auth.loginLoading);
   const loginError = useSelector((state: AppState) => state.auth.loginError);
   return (
-    <ImageBackground style={styles.container} source={background}>
-      <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : 'height'} style={styles.body}>
-        <Image source={logo} />
-        <TextInput
-          style={styles.input}
-          onChangeText={handleUserChange}
-          onEndEditing={handleUserEndEditing}
-          onSubmitEditing={Keyboard.dismiss}
-          value={userValue}
-          placeholder="user"
-        />
-        {userError && <Text style={styles.error}>Invalid Email</Text>}
-        <TextInput
-          style={styles.input}
-          onChangeText={handlePassChange}
-          onEndEditing={handlePasswordEndEditing}
-          onSubmitEditing={Keyboard.dismiss}
-          value={passwordValue}
-          placeholder="password"
-          secureTextEntry
-        />
-        {passwordError && <Text style={styles.error}>Invalid Password</Text>}
-        <TouchableOpacity
-          disabled={!buttonEnabled}
-          style={[styles.button, !buttonEnabled && styles.disabledButton]}
-          onPress={onPress}>
-          <Text style={[styles.buttonText, !buttonEnabled && styles.disabledButtonText]}>LOG IN</Text>
-        </TouchableOpacity>
-        {loginError && <Text style={[styles.error, styles.padding]}>Invalid credentials</Text>}
-      </KeyboardAvoidingView>
-      <Text style={styles.footer}>Designed, developed and used by woloxers</Text>
-    </ImageBackground>
+    <WithLoader loading={loginLoading}>
```

Comment:
Just as a comment, you could do something like this for the App's initial loading